### DISCUSSION TITLE
how to generate a new ciphertext based on a few existing ciphertexts

### DISCUSSION THREAD

**Discussion by macknight** [tuneinsight/lattigo#327] [Q&A] [answered]
Hi,
Suppose we have 3 ciphertexts, each representing 10 elements in each ciphertext, then how can we create a new ciphertext whose size is 3 with i-th element retrieving each first element of the 3 ciphertexts?

for example, we have c1, c2, c3. c1 represents 10 elements, c2 represents 10 elements, c3 represents 10 elements.
then how can we create a c* with size 3, c*[0] retrieves the first element of c1, c*[1] retrieves the first element of c2, c*[2] retrieves the first element of c3.

BR

**Comment by Pro7ech** [comment, marked as answer]
You will have to do: 

Mask (multiply with a plaintext that encrypts 1 at the position to extract else zero) each c_i to create a new ciphertext c'_i which encrypts the value you want else zero. 

Then rotate each c'_i to move the value to the desires slot.

Add all c'_i together and rescale.

**Reply by macknight** [reply]
> You will have to do:
> 
> Mask (multiply with a plaintext that encrypts 1 at the position to extract else zero) each c_i to create a new ciphertext c'_i which encrypts the value you want else zero.
> 
> Then rotate each c'_i to move the value to the desires slot.
> 
> Add all c'_i together and rescale.

This looks amazing, but a part of this question is about how to shrink the slots of this ciphertext?
For example, eachi c_i contains 10 elements, so slots of each may be 16,
And the new ciphertext has 3 elements, so slots of the new ciphertext can be 4.
So how to shrink?
BR

**Comment by ChristianMct** [comment]
Converting this to a discussion since this is not a question about Lattigo, but about about SIMD FHE schemes.
